src/dataframerow/dataframerow.jl

```diff
@@ -1,85 +1,123 @@
 """
-    DataFrameRow{<:AbstractDataFrame}
+    DataFrameRow{<:AbstractDataFrame,<:AbstractIndex}
 
-A view of one row of an AbstractDataFrame.
+A view of one row of an `AbstractDataFrame`.
+
+A `DataFrameRow` is constructed with `view` or `getindex` when one row and a
+selection of columns are requested, or when iterating the result
+of the call to the [`eachrow`](@ref) function.
+
+A `DataFrameRow` supports the iteration interface and can therefore be passed to
+functions that expect a collection as an argument.
+
+Indexing is one-dimensional like specifying a column of a `DataFrame`.
+You can also access the data in a `DataFrameRow` using the `getproperty` and
+`setproperty!` functions and convert it to a `NamedTuple` using the `copy` function.
+
+It is possible to create a `DataFrameRow` with duplicate columns.
+All such columns will have a reference to the same entry in the parent `DataFrame`.
+
+If the selection of columns in a parent data frame is passed as `:` (a colon)
+then `DataFrameRow` will always have all columns from the parent,
+even if it is mutated.
```

Comment:
```suggestion
even if they are added or removed after its creation.
```
(Same for `DataFrameRow`.)